### PR TITLE
Shift drag cloning

### PR DESCRIPTION
**Pull request by raheelx**
Hi @JoshEngebretson 

This PR allows the users to clone objects in the Scene by holding down on Shift and Dragging the object. 

Let me know if you'd like anything changed! :)

### CODE REVIEW

File: Source/AtomicEditor/Editors/SceneEditor3D/Gizmo3D.cpp [hunk 1]
```diff
@@ -38,8 +38,9 @@ namespace AtomicEditor
 {
 
 
-Gizmo3D::Gizmo3D(Context* context) : Object(context),
-    dragging_(false)
+	Gizmo3D::Gizmo3D(Context* context) : Object(context),
```
Comment (JoshEngebretson) [inline Source/AtomicEditor/Editors/SceneEditor3D/Gizmo3D.cpp]: Looks like some white space got introduced here

File: Source/AtomicEditor/Editors/SceneEditor3D/Gizmo3D.cpp [hunk 2]
```diff
@@ -184,6 +185,16 @@ void Gizmo3D::Use()
     Ray cameraRay = view3D_->GetCameraRay();
     float scale = gizmoNode_->GetScale().x_;
 
+	// Clones an object when it's dragged while holding down Shift
+	bool dragShift = input->GetMouseButtonDown(MOUSEB_LEFT) && input->GetKeyDown(KEY_SHIFT);
+
```
Comment (JoshEngebretson) [inline Source/AtomicEditor/Editors/SceneEditor3D/Gizmo3D.cpp]: Some more whitespace issues, indent is 4 spaces (can be set in VS options)

File: Source/AtomicEditor/Editors/SceneEditor3D/Gizmo3D.cpp [hunk 3]
```diff
@@ -193,6 +204,11 @@ void Gizmo3D::Use()
             scene_->SendEvent(E_SCENEEDITEND);
             dragging_ = false;
         }
+		
+		if (cloning_)
```
Comment (JoshEngebretson) [inline Source/AtomicEditor/Editors/SceneEditor3D/Gizmo3D.cpp]: Can just set cloning_ to false here without check, same result :)

**Comment by JoshEngebretson** [comment]
Nice!  Another great feature addition, and good job keeping the code tight.  There are some whitespace issues to address and one nit, otherwise ready to land.

**Comment by raheelx** [comment]
@JoshEngebretson : Sorry about the white spaces! I just started working on a new machine and forgot to change the settings :)

**Comment by JoshEngebretson** [comment]
No problem, great add, thanks!

LANDING! :airplane:

**Pull request merged**